snapcraft/internal/lxd/_project.py

```diff
@@ -112,10 +112,10 @@ def clean(self, parts: List[str], step: str):
                     print('Deleting {}'.format(self._container_name))
                     subprocess.check_call([
                         'lxc', 'delete', '-f', self._container_name])
-            step = 'pull'
+            step = steps.PULL.name
```

Comment:
I see two options here, but one thing first, we should rename the signature of this method to taking `step_name` instead of `step` or have it pass down an actual `Step`, where a possible value is `None` and move the DEPRECATED warning a level up when using `strip` passing it the appropriate step.

If using the former, I wouldn't get the name just to get the step again, but instead, only create the step for the case of `strip`

Reply:
Agreed, fixed by passing down a Step.